src/app/containers/Caption/index.jsx

```diff
@@ -10,40 +11,61 @@ import InlineLink from '../InlineLink';
 
 const componentsToRender = { fragment: Fragment, urlLink: InlineLink };
 
+const chooseOffscreenText = (mediaType, videoCaption, imageCaption) => {
+  let offscreenText = '';
+  if (mediaType === 'video') {
+    offscreenText = videoCaption;
+  } else if (mediaType === 'image') {
+    offscreenText = imageCaption;
+  } else {
+    offscreenText = false;
+  }
+  return offscreenText;
+};
+
 const renderText = textBlocks => (
   <Blocks blocks={textBlocks} componentsToRender={componentsToRender} />
 );
 
-const renderCaption = (block, imageCaptionOffscreenText) => (
-  <Caption>
-    {imageCaptionOffscreenText ? (
-      <VisuallyHiddenText>{imageCaptionOffscreenText}</VisuallyHiddenText>
+const renderCaption = (block, captionOffscreenText, key) => (
+  <Caption key={key}>
+    {captionOffscreenText ? (
+      <VisuallyHiddenText>{captionOffscreenText}</VisuallyHiddenText>
     ) : null}
     {renderText(block)}
   </Caption>
 );
 
-const renderMultipleCaptions = (blocks, imageCaptionOffscreenText) =>
-  blocks.map(block =>
-    renderCaption(
-      deepGet(['model', 'blocks'], block),
-      imageCaptionOffscreenText,
-    ),
+const renderMultipleCaptions = (blocks, type) => {
+  const {
+    videoCaptionOffscreenText,
+    imageCaptionOffscreenText,
+  } = React.useContext(ServiceContext);
+
+  const offscreenText = chooseOffscreenText(
+    type,
+    videoCaptionOffscreenText,
+    imageCaptionOffscreenText,
   );
 
-const CaptionContainer = ({ block }) => (
-  <ServiceContext.Consumer>
-    {({ imageCaptionOffscreenText }) =>
-      renderMultipleCaptions(
-        deepGet(['model', 'blocks', 0, 'model', 'blocks'], block),
-        imageCaptionOffscreenText,
-      )
-    }
-  </ServiceContext.Consumer>
-);
+  return blocks.map(block =>
+    renderCaption(deepGet(['model', 'blocks'], block), offscreenText, nanoid()),
```

Comment:
I know that `nanoid` is currently used in the codebase as a generator of unique ids. However recently there has been an issue opened regarding nanoid being terrible for performance and that it should be removed from our application.  https://github.com/bbc/simorgh/issues/1489  

Another way to generate a unique key value would be to use part of the content of the caption text.

Reply:
@sareh Would it be okay to do a deepGet just to get the caption text?

Reply:
@oluoluoxenfree Yes, that would be a good use of `deepGet`.